core/src/Cabana_AoSoA.hpp

```diff
@@ -126,27 +126,44 @@ class AoSoA
     /*!
       \brief Default constructor.
 
+      \param label An optional label for the data structure.
+
       The container size is zero and no memory is allocated.
     */
-    AoSoA()
+    explicit AoSoA( const std::string& label = "" )
         : _size( 0 )
         , _capacity( 0 )
         , _num_soa( 0 )
+        , _data( label )
     {}
 
     /*!
       \brief Allocate a container with n tuples.
 
       \param n The number of tuples in the container.
+
+      \param label An optional label for the data structure.
     */
-    explicit AoSoA( const int n )
+    AoSoA( const int n, const std::string label = "" )
```

Comment:
Change the order of the parameters here to follow kokkos conventions. Add back the original single size constructor for backwards compatibility and deprecate it.